### PR TITLE
Model in request on journey action

### PR DESCRIPTION
**Pull request by bambuchaAdm**
Add `ExportCacheModel` allow us to reduce number of database lookups. We are quering for same data multiple times:
1. In JourneyAction
2. In constroller it self

This change has intention to attached to request `ExportCacheModel` which was fetched in JourneyAction.

Also it fix stubbing for specific `sessionId` in cache mocks. It is still work in progress - i'm working on fixing tests.

### CODE REVIEW

File: app/controllers/declaration/AdditionalFiscalReferencesController.scala [hunk 1]
```diff
@@ -47,8 +47,8 @@ class AdditionalFiscalReferencesController @Inject()(
 )(implicit appConfig: AppConfig, ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable with SessionIdAware {
 
-  def displayPage(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    exportsCacheService.getItemByIdAndSession(itemId, journeySessionId).map {
+  def displayPage(itemId: String): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
+    request.cacheModel.itemBy(itemId) match {
```
Comment (ed0906) [inline app/controllers/declaration/AdditionalFiscalReferencesController.scala]: Nice

File: app/services/cache/ExportsCacheModelRepository.scala [hunk 2]
```diff
@@ -96,7 +96,9 @@ case class ExportsCacheModel(
   previousDocuments: Option[PreviousDocumentsData] = None,
   natureOfTransaction: Option[NatureOfTransaction] = None,
   seals: Seq[Seal] = Seq.empty
-)
+) {
+  def itemBy(itemId: String): Option[ExportItem] = items.find(_.id.equalsIgnoreCase(itemId))
```
Comment (ed0906) [inline app/services/cache/ExportsCacheModelRepository.scala]: 👍

File: app/services/cache/ExportsCacheService.scala [hunk 3]
```diff
@@ -30,9 +30,10 @@ class ExportsCacheService @Inject()(journeyCacheModelRepo: ExportsCacheModelRepo
   def update(sessionId: String, model: ExportsCacheModel): Future[Option[ExportsCacheModel]] =
     journeyCacheModelRepo.upsert(sessionId, model.copy(updatedDateTime = now()))
 
+  @deprecated("Please use `get` and `ExportCacheModel#itemBy` methods")
   def getItemByIdAndSession(itemId: String, sessionId: String): Future[Option[ExportItem]] =
```
Comment (ed0906) [inline app/services/cache/ExportsCacheService.scala]: Are there any usages?
Cant we just change them?

File: test/base/CustomExportsBaseSpec.scala [hunk 4]
```diff
@@ -192,17 +196,17 @@ trait CustomExportsBaseSpec
   }
 
   def withNewCaching(dataToReturn: ExportsCacheModel) {
-    when(mockExportsCacheService.getItemByIdAndSession(any[String], any[String]))
+    when(mockExportsCacheService.getItemByIdAndSession(any[String], ArgumentMatchers.eq(dataToReturn.sessionId)))
```
Comment (ed0906) [inline test/base/CustomExportsBaseSpec.scala]: Can we import `ArgumentMatchers._` to get rid of some of the clutter :)

File: app/controllers/actions/JourneyAction.scala [hunk 5]
```diff
@@ -41,8 +40,8 @@ case class JourneyAction @Inject()(cacheService: ExportsCacheService, mcc: Messa
     implicit val hc: HeaderCarrier =
       HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-    cacheService.get(request.session.data("sessionId")).map(_.map(_.choice)).map {
-      case Some(journeyType) => Right(JourneyRequest(request, Choice(journeyType)))
+    cacheService.get(request.sessionId).map {
+      case Some(cacheModel) => Right(JourneyRequest(request, cacheModel))
       case _ =>
         logger.error(s"Could not obtain journey type for ${eoriCacheId()(request)}")
         Left(Conflict("Could not obtain information about journey type"))
```
Comment (ed0906) [inline app/controllers/actions/JourneyAction.scala]: I know you didnt change this, but why on earth do we return a Conflict here....

It should be a redirect to the start page or something, but not a Conflict!

File: app/services/cache/ExportsCacheService.scala (hunk 3 above)
Comment (bambuchaAdm) [reply app/services/cache/ExportsCacheService.scala]: I use `deprecated` in development to find all changes. I will remove this before merge.

File: app/controllers/actions/JourneyAction.scala (hunk 5 above)
Comment (PatrykRudnicki) [reply app/controllers/actions/JourneyAction.scala]: Magic 409 😂

**Comment by bambuchaAdm** [comment]
please build this

**Comment by bambuchaAdm** [comment]
build this please

**Pull request closed without merging**